plugins/providers/bigquery/provider.go

```diff
@@ -3,28 +3,47 @@ package bigquery
 import (
 	"context"
 	"errors"
+	"fmt"
 	"strings"
 
+	bq "cloud.google.com/go/bigquery"
 	"github.com/mitchellh/mapstructure"
 	"github.com/odpf/guardian/core/provider"
 	"github.com/odpf/guardian/domain"
 )
 
+//go:generate mockery --name=BigQueryClient --exported --with-expecter
+type BigQueryClient interface {
+	GetDatasets(context.Context) ([]*Dataset, error)
+	GetTables(ctx context.Context, datasetID string) ([]*Table, error)
+	GrantDatasetAccess(ctx context.Context, d *Dataset, user, role string) error
+	RevokeDatasetAccess(ctx context.Context, d *Dataset, user, role string) error
+	GrantTableAccess(ctx context.Context, t *Table, accountType, accountID, role string) error
+	RevokeTableAccess(ctx context.Context, t *Table, accountType, accountID, role string) error
+	ResolveDatasetRole(role string) (bq.AccessRole, error)
+	ListAccess(ctx context.Context, resources []*domain.Resource) (domain.ResourceAccess, error)
+}
+
+//go:generate mockery --name=crypto --exported --with-expecter
+type crypto interface {
```

Comment:
perhaps `Encryptor` defines it better for interface (although it might do encryption and depcryption)

https://go.dev/doc/effective_go#interface-names

Reply:
renamed it to encryptor in the bigquery package